webform_civicrm_membership_extras.module

```diff
@@ -21,6 +21,474 @@ function webform_civicrm_membership_extras_webform_submission_insert($node, $sub
     $installmentsHandler = new CRM_MembershipExtras_Service_MembershipInstallmentsHandler($contributionRecurId);
     $installmentsHandler->createRemainingInstalmentContributionsUpfront();
   }
+
+  $discountCodeDetails = _membershipextras_getSubmittedDiscountCodeDetails($node, $submission);
+  if ($discountCodeDetails) {
+    _membershipextras_updateDiscountCodeUsage($discountCodeDetails, $node);
+  }
+}
+
+/**
+ * Implements hook_civicrm_apiWrappers().
+ *
+ * @param $wrappers
+ * @param $apiRequest
+ */
+function webform_civicrm_membership_extras_civicrm_apiWrappers(&$wrappers, $apiRequest) {
+  // Add custom civicrm api wrapper to alter line item price
+  $wrappers[] = new wf_me_discount_civicrm_apiwrapper();
+}
+
+
+/**
+ * Implements hook_form_alter().
+ */
+function webform_civicrm_membership_extras_form_alter(&$form, &$form_state, $form_id) {
+  $wf_me_cividiscount_helper = new wf_me_cividiscount_ext_helper();
```

Comment:
This method is very long also. Do you think you can clean it up a little, by breaking it into other methods? Perhaps, at least one per form_id. Maybe even classes, if you think it makes sense.